### ISSUE TITLE
Asset naming for deployment (js/png/gif/jpg/etc)

### ISSUE THREAD

**Issue by NoralK** [SharePoint/sp-dev-fx-webparts#309]
#### Category
- [X] Question
- [ ] Bug
- [ ] Enhancement

#### Expected or Desired Behavior
Is there the ability to have statically named assets or the ability to rename the assets in the configuration files during the build process?  Right now I am performing this task manually after I bundle the shipment but before I package-solution.

#### Observed Behavior
Assets are being renamed to unwanted long names.

#### Steps to Reproduce
Create new project
Gulp build -ship
Gulp bundle -ship

Look at files in \temp\deploy
The js, images and, I assume other assets, if added, are being renamed.  Here are example file names:

**Current files**
2ea3efb6-59b1-499f-94c2-9af2eabe36e2.manifest.json
logo_e5e94bb4b4e8155255b2b654568170dd.png
my-on-prem-web-part.bundle_e8a71d1a85c8f03b74e2193824de14bb.js
mywebpart1-myonpremwebpartstrings_en-us_536e65149b0acf4d52c0043073b9fc59.js

**Desired named files**
my-on-prem-web-part.manifest.json
logo.png
my-on-prem-web-part.bundle.js
mywebpart1-myonpremwebpartstrings_en-us.js

When deploying I am putting each version of the WebPart in a separate sub-directory, in our CDN, for each version so I will not have duplicate files being clobbered.

TIA
Noral

**Comment by waldekmastykarz** [comment]
Next to avoiding name duplication, another reason, for which you might want to keep these unique names, is that it enables you to leverage far-future caching. You could have users store your web part assets in their local cache for a year, so if they're using your web part daily, they would download it only once, which would significantly improve performance. Because you can't automatically invalidate files cached this way, using different names when deploying new versions, allows you to force users to load new versions of the assets.

**Comment by NoralK** [comment]
Thank you for the information.  I can see letting the JavaScript be uniquely named, however would there be a way, like I mentioned, to name them or keep their original names?

**Comment by waldekmastykarz** [comment]
I don't think there is currently a robust way to do that.

**Comment by bratukham** [comment]
@NoralK There is a possibility to override webpack config. You probably can achieve your goals doing this, but it will be a bit tricky. On of the examples you can find at the end of this article: https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/use-fabric-react-components

**Comment by spdevdocs** [comment]
We closed this issue as it had not activity within last 180 days. This is a generic process we have decided to perform for issues, which have not been explicitly marked still to be "work in progress" based on tags. We are performing this cleaning to make sure that old issues that have already been solved (but not closed) or are no longer relevant are cleaned out and make the issues more manageable. If this issue still valid, we would ask you to **open a new issue** and follow the guidance in the issue template related on the recommended location. We do apologize any inconveniences this might cause. Please do remember that issues in the issue lists are also messages for others in the community, so you can also check if you can assist on any of them. *“Sharing is caring!”*
